### PR TITLE
Refactor extensions

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/components/dialogs/network-modifications/voltage-level/creation/voltage-level-creation-dialog.js [hunk 1]
```diff
@@ -147,10 +147,14 @@ const VoltageLevelCreationDialog = ({
                 [LOW_VOLTAGE_LIMIT]: voltageLevel[LOW_VOLTAGE_LIMIT],
                 [HIGH_VOLTAGE_LIMIT]: voltageLevel[HIGH_VOLTAGE_LIMIT],
                 [LOW_SHORT_CIRCUIT_CURRENT_LIMIT]: unitToKiloUnit(
-                    voltageLevel.ipMin
+                    fromCopy
+                        ? voltageLevel?.identifiableShortCircuit?.ipMin
+                        : voltageLevel.ipMin
```
Comment (SlimaneAmar) [inline src/components/dialogs/network-modifications/voltage-level/creation/voltage-level-creation-dialog.js:152]: ?
without extension

File: src/components/spreadsheet/utils/config-tables.js [hunk 2]
```diff
@@ -179,33 +183,53 @@ export const TABLES_DEFINITIONS = {
...
-                'ipMax',
-                1,
-                'equipment.setIpMax({})\n',
-                false,
-                'ipMin',
-                undefined,
-                excludeFromGlobalFilter,
-                {
+            {
+                id: 'IpMin',
```
Comment (SlimaneAmar) [inline src/components/spreadsheet/utils/config-tables.js:187]: Why we not use functions ?

File: src/components/dialogs/network-modifications/voltage-level/creation/voltage-level-creation-dialog.js (hunk 1 above)
Comment (souissimai) [reply src/components/dialogs/network-modifications/voltage-level/creation/voltage-level-creation-dialog.js:152]: Yes, It depends on the formCopy if it's true we are getting infos from network-map-server else it's from network-modification ( the data model in network did not change)

File: src/components/spreadsheet/utils/config-tables.js (hunk 2 above)
Comment (souissimai) [reply src/components/spreadsheet/utils/config-tables.js:187]: generateEditableNumericColumnDefinition uses cellEditorParams, but in this case we need valueGetter and valueSetter

File: src/components/dialogs/network-modifications/battery/creation/battery-creation-dialog.js [hunk 3]
```diff
@@ -111,8 +111,9 @@ const BatteryCreationDialog = ({
             [MINIMUM_ACTIVE_POWER]: battery.minP,
             [ACTIVE_POWER_SET_POINT]: battery.targetP,
             [REACTIVE_POWER_SET_POINT]: battery.targetQ,
-            [FREQUENCY_REGULATION]: battery.activePowerControlOn,
-            [DROOP]: battery.droop,
+            [FREQUENCY_REGULATION]:
+                battery?.activePowerControl?.activePowerControlOn,
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/components/dialogs/network-modifications/battery/creation/battery-creation-dialog.js]: done

**Review by Tristan-WorkGH — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**